### PR TITLE
Adding a filter for SkipStatusCodePages and adding a sample

### PR DESCRIPTION
**Pull request by Praburaj**
...pages

@yishaigalatzer

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    /// Filter to prevent StatusCodePages middleware to handle responses.
+    /// </summary>
+    public class SkipStatusCodePagesAttribute : ResultFilterAttribute
+    {
+        public override void OnResultExecuted(ResultExecutedContext context)
+        {
+            var statusCodeFeature = context.HttpContext.GetFeature<IStatusCodePagesFeature>();
+            if (statusCodeFeature != null)
+            {
+                // Turns off the StatusCodePages feature.
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs]: > Turn off the status code page

File: src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using Microsoft.AspNet.Diagnostics;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Filter to prevent StatusCodePages middleware to handle responses.
+    /// </summary>
+    public class SkipStatusCodePagesAttribute : ResultFilterAttribute
+    {
+        public override void OnResultExecuted(ResultExecutedContext context)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs]: > /// `<inheriteddoc />`

**Comment by pranavkm** [comment]
Needs unit tests for the attribute.

**Comment by pranavkm** [comment]
:watch:

**Comment by Praburaj** [comment]
@pranavkm fixed and updated.

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            var statusCodePagesFeature = new TestStatusCodeFeature();
+            resultExecutingContext.HttpContext.SetFeature<IStatusCodePagesFeature>(statusCodePagesFeature);
+            var next = new ResultExecutionDelegate(
+                            () => Task.FromResult(CreateResultExecutedContext(resultExecutingContext)));
+
+            // Act
+            await skipStatusCodeAttribute.OnResultExecutionAsync(resultExecutingContext, next);
+
+            // Assert
+            var objectResult = resultExecutingContext.Result as ObjectResult;
```
Comment (Praburaj) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: I'll remove this. I don't need this line.

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            // Arrange
+            var skipStatusCodeAttribute = new SkipStatusCodePagesAttribute();
+            var resultExecutingContext = CreateResultExecutingContext(new IFilter[] { skipStatusCodeAttribute });
+            var statusCodePagesFeature = new TestStatusCodeFeature();
+            resultExecutingContext.HttpContext.SetFeature<IStatusCodePagesFeature>(statusCodePagesFeature);
+            var next = new ResultExecutionDelegate(
+                            () => Task.FromResult(CreateResultExecutedContext(resultExecutingContext)));
+
+            // Act
+            await skipStatusCodeAttribute.OnResultExecutionAsync(resultExecutingContext, next);
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: Couldn't you run the `OnResultExecuted` instead?

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+            // Assert
+            Assert.Equal(false, statusCodePagesFeature.Enabled);
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: `Assert.False`

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        private static ResultExecutedContext CreateResultExecutedContext(ResultExecutingContext context)
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: Need another test to verify it no-ops \ doesn't null ref if the feature is missing.

**Comment by pranavkm** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs (hunk 4 above)
Comment (Praburaj) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: The filter executes `OnResultExecuted`. I saw some samples around and thought this is the way. I can change to:

```
// Arrange
            var skipStatusCodeAttribute = new SkipStatusCodePagesAttribute();
            var resultExecutingContext = CreateResultExecutingContext(new IFilter[] { skipStatusCodeAttribute });
            var statusCodePagesFeature = new TestStatusCodeFeature();
            resultExecutingContext.HttpContext.SetFeature<IStatusCodePagesFeature>(statusCodePagesFeature);

            // Act
            skipStatusCodeAttribute.OnResultExecuted(CreateResultExecutedContext(resultExecutingContext));
```

File: test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs (hunk 6 above)
Comment (Praburaj) [inline test/Microsoft.AspNet.Mvc.Core.Test/Filters/SkipStatusCodePagesAttributeTest.cs]: sure.

File: src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Diagnostics;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Filter to prevent StatusCodePages middleware to handle responses.
+    /// </summary>
+    public class SkipStatusCodePagesAttribute : ResultFilterAttribute
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs]: this should be a resource filter, you basically signal that this action is not participating in status pages.

Resource filters always run (as long as you passed authorization), result filters however don't always run, for example if an action filter short circuited the result

File: src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        public override void OnResultExecuted(ResultExecutedContext context)
+        {
+            var statusCodeFeature = context.HttpContext.GetFeature<IStatusCodePagesFeature>();
+            if (statusCodeFeature != null)
+            {
+                // Turn off the StatusCodePages feature.
+                statusCodeFeature.Enabled = false;
+            }
+
+            base.OnResultExecuted(context);
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs]: you don't need to call base in filters

File: src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs (hunk 7 above)
Comment (Praburaj) [inline src/Microsoft.AspNet.Mvc.Core/Filters/SkipStatusCodePagesAttribute.cs]: I see. I'll send out a PR for addressing this.
